packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -241,9 +243,13 @@ function SelectControl< ItemType = DefaultItemType >( {
 	} );
 
 	const isEventOutside = ( event: React.FocusEvent ) => {
-		return ! document
-			.querySelector( '.' + instanceId )
-			?.contains( event.relatedTarget );
+		const inputClasses = event?.target?.className;
+		return (
+			! document
+				.querySelector( '.' + instanceId )
+				?.contains( event.relatedTarget ) &&
+			! inputClasses.includes( listItemClassName )
```

Comment:
Does the original issue occur because of focusing the input?  I'm curious why this isn't under the instance.

Or is the issue that the popover list items are slotfilled into a div outside of the instance?  If that's the case, we probalby don't need to add `woocommerce-experimental-select-control__list-item` to the input below, right?

Reply:
I believe the problem originates from the element's `onBlur` method. When the list opens after an item's name is clicked, the `onBlur` method is triggered. This method calls ([isEventOutside](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/experimental-select-control/select-control.tsx#L243)) and it returns `true`, even though the click was actually made inside. This is why I added the class to have a more specific check.
An alternative approach could be to only use the class `woocommerce-experimental-select-control__input`, which is already in [use there](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/experimental-select-control/select-control.tsx#L300).
Does that sound good to you?

Reply:
I created [this commit](https://github.com/woocommerce/woocommerce/pull/39574/commits/57f2879ccc1ee05165f56e775a8c6ca0fce5b911) that removes the other class name and uses `woocommerce-experimental-select-control__input` to verify that the click has been done on the input in the list.